cmd/deployIntegrationArtifact_test.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	cpi "github.com/SAP/jenkins-library/pkg/cpi"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type deployIntegrationArtifactMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newDeployIntegrationArtifactTestsUtils() deployIntegrationArtifactMockUtils {
+	utils := deployIntegrationArtifactMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunDeployIntegrationArtifact(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Integration Flow Deploy Test", func(t *testing.T) {
+		// init
+
+		config := deployIntegrationArtifactOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			Platform:               "cf",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "DeployIntegrationDesigntimeArtifact", StatusCode: 202,
+			ResponseBody: ``, TestType: "Positive"}
+
+		err := runDeployIntegrationArtifact(&config, nil, &httpClient)
+		// assert
+		assert.NoError(t, err)
+	})
+
+	t.Run("Failed case of Integration Flow Deploy Test", func(t *testing.T) {
+		// init
+		config := deployIntegrationArtifactOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			Platform:               "cf",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "DeployIntegrationDesigntimeArtifact", StatusCode: 202,
+			ResponseBody: ``, TestType: "Negative"}
+
+		err := runDeployIntegrationArtifact(&config, nil, &httpClient)
+		// assert
+		assert.EqualError(t, err, "Integration Flow deployment failed")
+	})
+
+}
+
+type httpMockCpis struct {
+	Method       string                  // is set during test execution
+	URL          string                  // is set before test execution
+	Header       map[string][]string     // is set before test execution
+	ResponseBody string                  // is set before test execution
+	Options      piperhttp.ClientOptions // is set during test
+	StatusCode   int                     // is set during test
+	CPIFunction  string                  // is set during test
+	TestType     string                  // is set during test
+}
+
+func (c *httpMockCpis) SetOptions(options piperhttp.ClientOptions) {
+	c.Options = options
+}
+
+func (c *httpMockCpis) SendRequest(method string, url string, r io.Reader, header http.Header, cookies []*http.Cookie) (*http.Response, error) {
```

Comment:
Now, as `r` is not used anymore I would like to suggest to rename it to `_` in order to outline that this is not used. Of course we cannot remove that parameter from the signature since with that we would not implement the interface. `_` is the go idiom for variables which are not used.

Reply:
I thought about it and I changed my mind. Reading `r` mimics the behaviour of the non-mock instance which we have in real life. In real like the request body - which is provided with `r` will be read. Maybe some later code flow will rely on the fact that `r` has been read. From that point of view I now think it is fine to have that lines reading `r`.

Reply:
i added it back